spack/packages/pumgen/package.py

```diff
@@ -8,30 +8,42 @@
 
 class Pumgen(CMakePackage):
     homepage = "https://github.com/SeisSol/PUMGen/wiki/How-to-compile-PUMGen"
+    git = "https://github.com/SeisSol/PUMGen.git"
     version('develop',
-            git='https://github.com/SeisSol/PUMGen.git',
             branch='master',
             submodules=True)
 
+    version('1.0.1', tag='v1.0.1', commit='13d1e1f41e3cdcb1d3c94f1870f003778e5f0ce0', submodules=True)
     maintainers = ['ravil-mobile']
     variant('with_simmetrix', default=False)
+    variant('scorec', default=False)
     variant('with_netcdf', default=False)
 
     depends_on('mpi')
-        
+    # simmetrix (e.g. @2024.0-240616) currently does not have a precompiled libSimPartitionWrapper-openmpi5.a 
+    depends_on(f"openmpi@:4", when=f"+with_simmetrix ^[virtuals=mpi] openmpi")
+
     depends_on('netcdf-c +shared +mpi', when='+with_netcdf') # NOTE: only tested with 4.4.0 version
-    depends_on('hdf5@1.10:1.12.2 +shared +threadsafe +mpi')
-    depends_on('pumi +int64 +zoltan ~fortran', when='~with_simmetrix')
+    depends_on('hdf5@1.10: +shared +threadsafe +mpi')
     depends_on('simmetrix-simmodsuite', when='+with_simmetrix')
-    depends_on('pumi +int64 simmodsuite=base +zoltan ~fortran ~simmodsuite_version_check', when='+with_simmetrix')
+    depends_on('pumi +int64 simmodsuite=base +zoltan ~fortran ~simmodsuite_version_check', when='+with_simmetrix +scorec')
+    with when("@1.0.1"):
```

Comment:
You could also go for a `requires("+scorec", when="@:1.0.1")` or similar, if you want to shorten the code a bit

Reply:
yes, but  this does not covers pumi without simmetrix for @1.0.1
(that means user would need to have a simmetrix license to install)